app/javascript/services/courses-service.js

```diff
@@ -0,0 +1,10 @@
+import axios from '../helpers/axios-rails';
+import {courseJoinRoute} from "./service-routes";
+
+class CoursesService {
+    static async joinCourse(courseId) {
+        return axios.post(courseJoinRoute(courseId)).then(response=>response.data);
+    }
+}
+
+export default CoursesService;
```

Comment:
I wonder whether this should be part of this PR, since it doesn't seem to related to the tests.